### DISCUSSION TITLE
Is there a plan to support index prop changes?

### DISCUSSION THREAD

**Discussion by stonesthatwhisper** [igordanchenko/yet-another-react-lightbox#74] [Ideas]
The doc for `index` property currently reads

> Changing this property while the lightbox is already open has no effect.

Is there any plan to make it capable of being controlled by a React state? That would enable some extra use cases YARL can implement. edit: I apologize for the truncated title. It seems not editable to me now.

**Comment by igordanchenko** [comment]
Can you describe the use cases you have in mind?

I've been thinking about externalizing the lightbox state, but it's not as trivial as it may sound (unless I force all users to maintain the current slide index, swipe state, swipe offset, animation duration, and some other animation-related bits of the lightbox state externally).

As I can see, the vast majority of public GitHub repositories use YARL the way I assumed it would be used in most cases.

- static slides array
- the lightbox opens at the slide indicated via `index` prop
- the user browses some slides and closes the lightbox

**Comment by igordanchenko** [comment]
@stonesthatwhisper 

I started drafting v3 to improve the current implementation. If you'd like your use case(s) to be considered, now is a good time to share 😉
